specs/src/tx_client.md

```diff
@@ -0,0 +1,254 @@
+                  ErrorLog string // Detailed error message
+                }
+                ```
+
+6. **Tracker Entry**: On success, record (`signer`, `sequence`, `txBytes`, `timestamp`) in TxTracker. Entries older than 10 minutes are pruned automatically.
+7. **Sequence Increment**: After successful broadcast, increment the signer’s local sequence by calling **signer.SetSequence()**
+
+### Confirmation
+
+After broadcast, the TxClient continuously polls the chain for transaction status until resolution. This is the tx confrimation life cycle:
+
+- **Pending**: Transaction in the mempool but not yet included, continue polling
+
+- **Committed**: Transaction included in a block.
+  - If execution succeeded (response code = `abci.CodeTypeOK`): return success
+  - If execution failed (any other response code):
+    - Populate `ExecutionError` and return it to the user
+
+    ```go
+    type ExecutionError struct {
+        TxHash   string // Transaction hash
+        Code     uint32 // Execution result code
+        ErrorLog string // Execution error details
+    }
+    ```
+
+- **Rejected**: Transaction was explicitly refused by the node during `ReCheck()` (after each block commit).
+  - Retrieve tx sequence from `TxTracker`
+  - Roll back the sequence to the retrieved tx sequence
+  - Construct and return a rejection error with the tx hash and response code. (no specific error type)
+
+- **Evicted**: Transaction dropped from mempool (low fees, mempool full).
+  - The client resubmits using txBytes stored in the tracker.
+  - If resubmission fails (e.g. sequence mismatch):
+    - **Do not resign with a new sequence**. Eviction isn't final, another node might still propose this tx and resigning risks duplicate inclusion.
+    - Start a 1-minute eviction timeout window. If the tx is not confirmed within timeout period, construct and return an eviction error. (no specific error type)
+  - If resubmission is successful:
+    - The same tx is re-added to the node's mempool
+    - Continue polling until the updated status is resolved
+
+- **Unknown/Not Found**: Tx is neither evicted nor rejected or confirmed. Return error to the user that tx is unknown.
+
+- Once transactions are concluded (rejected/evicted/confirmed) they should get pruned from the `TxTracker` before function returns.
+
+## Message Structure/Communication Format
+
+The client communicates with Celestia full nodes through gRPC services for transaction broadcasting, status checking, and gas estimation.
+
+### Transaction Broadcast
+
+```go
+Request: BroadcastTxRequest { TxBytes, Mode }
+Response: BroadcastTxResponse { TxHash, Code, RawLog }
+```
+
+### Transaction Status
+
+```go
+Request: TxStatusRequest { TxId }
+Response: TxStatusResponse { Status, Height, ExecutionCode, Error }
+```
+
+### Gas Estimation
+
+```go
+Request: EstimateGasPriceAndUsageRequest { TxBytes, TxPriority }
+Response: EstimatedGasUsed, EstimatedGasPrice
+```
```

Comment:
not blocking

don't we cover this on lines 43

Reply:
This is trying to break down communication format. I'm happy to remove though.